Multitrack-Template.lua

```diff
@@ -0,0 +1,205 @@
+
+-- Dictionary holding Key Value Pairs
+-- for input labels
+-- Key is derived from I don't know what yet
+local i = {};
+for n=(PLAYER_COUNT-1),0,-1 do
+	i[1+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Up";
+	i[2+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Down";
+	i[3+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Left";
+	i[4+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Right";
+	i[5+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Button 1";
+	i[6+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Button 2";
+	i[7+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Button 3";
+	i[8+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Start";
+	i[9+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Coin";
+end
+
+-- Max number of players
+PLAYER_COUNT = table.getn(NameTable);
+
+-- Total number of buttons being drawn in UI
+MONITORED_BUTTON_COUNT = table.getn(visualtbl);
+
+-- Table representing all the input buttons for your game
+INPUT_BUTTON_COUNT = table.getn(tabinp);
+
+current_pressed_keys, last_pressed_keys = {}, {};
+AllFrames = {};
+
+local inp ={};
+local StartFrame = emu.framecount();
+local RecFrames = 0;
+
+-- This is the character whose inputs are currently being recorded
+-- Toggling between characters is controlled by the test function - which looks
+-- at a special set of keys set aside to switch control between the playable
+-- characters
+character_being_recorded = 3;
+
+-- Print contents of `tbl`, with indentation.
+-- `indent` sets the initial level of indentation.
+function tprint (tbl, indent)
+  if not indent then indent = 0 end
+  for k, v in pairs(tbl) do
+    formatting = string.rep("  ", indent) .. k .. ": "
+    if type(v) == "table" then
+      print(formatting)
+      tprint(v, indent+1)
+    elseif type(v) == 'boolean' then
+      print(formatting .. tostring(v))
+    else
+      print(formatting .. v)
+    end
+  end
+end
+
+--*****************************************************************************
+function press(button)
+--*****************************************************************************
+-- Checks if a button is pressed.
+-- The tables it accesses should be obvious in the next line.
+
+    if current_pressed_keys[button] and not last_pressed_keys[button] then
+        return true
+
+    end
+    return false
+end
+
+function test()
+	current_pressed_keys = input.get();
+	--print(current_pressed_keys);
+	RF = {};
+	if current_pressed_keys['pageup'] then
+		character_being_recorded = 1;
+	elseif current_pressed_keys['pagedown'] then
+		character_being_recorded = 2;
+	elseif current_pressed_keys['numpad3'] then
+		character_being_recorded = 3;
+	elseif current_pressed_keys['numpad4'] then
+		character_being_recorded = 4;
+	end;
+
+  --print(movie.mode());
+	if movie.mode() ~= 'playback' then
+  --print('WE ARE HERE');
+		-- Loop through characters
+		for n = (PLAYER_COUNT-1),0,-1 do
+			--print("CURRENT CHARACTER "..(character_being_recorded).. "=="..(n+1));
+			if character_being_recorded == n+1 or character_being_recorded == 4 then
```

Comment:
This logic could be refactored

note it's currently hacked/hardcoded to my own TAS needs at the moment